test-suite/Math/NumberTheory/ArithmeticFunctionsTests.hs

```diff
@@ -270,6 +272,45 @@ mangoldtOeis = oeisAssertion "A014963" expMangoldtA
   , 73, 1, 1, 1, 1, 1, 79, 1, 3, 1, 83, 1, 1, 1, 1, 1, 89, 1, 1, 1, 1, 1, 1
   ]
 
+nFreedomProperty1 :: Word -> NonZero Natural -> Bool
+nFreedomProperty1 n (NonZero m) =
+    isNFree n m == (all ((< n) . snd) . factorise) m
+
+nFreedomProperty2 :: NonZero Word -> NonNegative Int -> Bool
```

Comment:
There is `Math.NumberTheory.TestUtils.Wrappers.Power`, which seems more suitable than `NonZero` here.